time/src/main/java/net/jqwik/time/DefaultMonthDayArbitrary.java

```diff
@@ -0,0 +1,80 @@
+package net.jqwik.time;
+
+import java.time.*;
+
+import net.jqwik.api.*;
+import net.jqwik.api.arbitraries.*;
+import net.jqwik.api.time.*;
+
+public class DefaultMonthDayArbitrary extends ArbitraryDecorator<MonthDay> implements MonthDayArbitrary {
+
+	private MonthDay monthDayMin = MonthDay.of(Month.JANUARY, 1);
+	private MonthDay monthDayMax = MonthDay.of(Month.DECEMBER, 31);
+	private Month monthMin = Month.JANUARY;
+	private Month monthMax = Month.DECEMBER;
+	private Month[] allowedMonths = new Month[]{Month.JANUARY, Month.FEBRUARY, Month.MARCH, Month.APRIL, Month.MAY, Month.JUNE, Month.JULY, Month.AUGUST, Month.SEPTEMBER, Month.OCTOBER, Month.NOVEMBER, Month.DECEMBER};
+	private int dayOfMonthMin = 1;
+	private int dayOfMonthMax = 31;
+
+
+	@Override
+	protected Arbitrary<MonthDay> arbitrary() {
+		DateArbitrary dates = Dates.dates()
+								   .atTheEarliest(LocalDate.of(0, monthDayMin.getMonth(), monthDayMin.getDayOfMonth()))
+								   .atTheLatest(LocalDate.of(0, monthDayMax.getMonth(), monthDayMax.getDayOfMonth()))
+								   .yearBetween(0, 0)
+								   .monthBetween(monthMin, monthMax)
+								   .onlyMonths(allowedMonths)
+								   .dayOfMonthBetween(dayOfMonthMin, dayOfMonthMax);
+		return dates.map(v -> MonthDay.of(v.getMonth(), v.getDayOfMonth()));
+	}
```

Comment:
this looks very complicated for "only" creating a valid date within a year.
Can year 0 be a lap year? Should Feb 29 be possible.

Reply:
Yes, year 0 can be a lap year. Should I change it? I thought it is good to use existing code instead of writing a completely new arbitrary.

Reply:
> Yes, year 0 can be a lap year. Should I change it? I thought it is good to use existing code instead of writing a completely new arbitrary.

Reuse is good... if it doesn't complicate things. Generating dates and stripping away the year will lead to bad performance concerning shrinking and exhaustive genration and lead to duplicate edge cases.

Reply:
even if the year is always 0?

Reply:
Not sure. Have you done an experiment to check?

Reply:
No, but `Dates.years().between(yearMin, yearMax)` creates only one year wit the value 0. Only `generateMonths` and `generateDayOfMonths` have more values. So the year value is always the same and can't change.